webclient/grpc/src/main/java/io/helidon/webclient/grpc/ClientUriSuppliers.java

```diff
@@ -0,0 +1,210 @@
+        }
+
+        /**
+         * Creates a round-robin supplier.
+         *
+         * @param clientUris collection of client URIs
+         * @return new supplier
+         */
+        public static RoundRobinSupplier create(Collection<ClientUri> clientUris) {
+            return new RoundRobinSupplier(clientUris.toArray(new ClientUri[]{}));
+        }
+
+        protected RoundRobinSupplier(ClientUri[] clientUris) {
+            this.clientUris = clientUris;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return true;
+        }
+
+        @Override
+        public ClientUri next() {
+            return clientUris[next++ % clientUris.length];
+        }
+    }
+
+    /**
+     * Supplies the same client URI over and over, never ends.
+     */
+    public static class SingleSupplier implements ClientUriSupplier {
+
+        private final ClientUri clientUri;
+
+        /**
+         * Creates a single supplier.
+         *
+         * @param clientUri the client URI as a string
+         * @return new supplier
+         */
+        public static SingleSupplier create(String clientUri) {
+            return new SingleSupplier(ClientUri.create(URI.create(clientUri)));
+        }
+
+        /**
+         * Creates a single supplier.
+         *
+         * @param clientUri the client URI
+         * @return new supplier
+         */
+        public static SingleSupplier create(ClientUri clientUri) {
+            return new SingleSupplier(clientUri);
+        }
+
+        protected SingleSupplier(ClientUri clientUri) {
+            this.clientUri = clientUri;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return true;
+        }
+
+        @Override
+        public ClientUri next() {
+            return clientUri;
+        }
+    }
+
+    /**
+     * Supplies an iterator that returns a URI chosen at random.
+     */
+    public static class RandomSupplier implements ClientUriSupplier {
+
+        private boolean supplied;
+        private final ClientUri[] clientUris;
+        private final SecureRandom random = new SecureRandom();
+
+
+        /**
+         * Creates a random supplier.
+         *
+         * @param clientUris array of client URIs
+         * @return new supplier
+         */
+        public static RandomSupplier create(ClientUri... clientUris) {
+            return new RandomSupplier(clientUris);
+        }
+
+        /**
+         * Creates a random supplier.
+         *
+         * @param clientUris collection of client URIs
+         * @return new supplier
+         */
+        public static RandomSupplier create(Collection<ClientUri> clientUris) {
+            return new RandomSupplier(clientUris.toArray(new ClientUri[]{}));
+        }
+
+        protected RandomSupplier(ClientUri[] clientUris) {
+            this.clientUris = clientUris;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return supplied;
```

Comment:
I see the code that uses `ClientUriSupplier` does not actually call `hasNext()` but this will return `false` if `next()` has not been called and only after `next()` is called will it return `true`, but then calling `next()` again will throw `NoSuchElementException`. I think this should be `return !supplied`

Reply:
Yes, bug fixed